tasks/ebpf.py

```diff
@@ -0,0 +1,125 @@
+try:
+    from termcolor import colored
+except ImportError:
+    colored = None
+import contextlib
+import json
+import sys
+
+from invoke import task
+
+try:
+    from tabulate import tabulate
+except ImportError:
+    tabulate = None
+
+from .system_probe import build_object_files, is_root
+
+headers = [
+    "Program",
+    "Stack Usage",
+    "Instructions Processed",
+    "Instructions Processed limit",
+    "Verification time",
+    "Max States per Instruction",
+    "Peak States",
+    "Total States",
+]
+
+
+def tabulate_stats(stats):
+    table = list()
+    for key, value in stats.items():
+        row = list()
+        row.append(key)
+        row.append(value["stack_usage"])
+        row.append(value["instruction_processed"])
+        row.append(value["limit"])
+        row.append(value["verification_time"])
+        row.append(value["max_states_per_insn"])
+        row.append(value["peak_states"])
+        row.append(value["total_states"])
+        table.append(row)
+
+    return tabulate(table, headers=headers, tablefmt="github")
+
+
+def colored_diff(val1, val2):
+    if val1 <= val2:
+        return colored(val1 - val2, "green")
+    return colored(val1 - val2, "red")
+
+
+def stdout_or_file(filename=None):
+    @contextlib.contextmanager
+    def stdout():
+        yield sys.stdout
+
+    return open(filename, 'w') if filename else stdout()
+
+
+@task(
+    help={
+        "skip_object_files": "Do not build ebpf object files",
+        "base": "JSON file holding verifier statistics to compare against",
+        "jsonfmt": "Output in json format rather than tabulating",
+        "out": "Output file to write results to. By default results are written to stdout",
+    }
+)
+def print_verification_stats(ctx, skip_object_files=False, base=None, jsonfmt=False, out=None, debug_build=False):
+    sudo = "sudo" if not is_root() else ""
+    if not skip_object_files:
+        build_object_files(ctx)
+
+    # generate programs.go
+    use_debug_build = "USE_DEBUG_BUILDS='true' " if debug_build else ""
+    ctx.run(f"{use_debug_build}go generate pkg/ebpf/verifier/stats.go")
+
+    ctx.run("cd pkg/ebpf/verifier && go generate")
+    ctx.run("go build -tags linux_bpf pkg/ebpf/verifier/calculator/main.go")
+
+    debug = "--debug" if debug_build else ""
+    res = ctx.run(f"{sudo} ./main --directory pkg/ebpf/bytecode/build/co-re {debug}", hide=True)
+    print(res.stderr)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Was this print left on purpose ? If that's the case maybe we should put it in python stderr ? Smth like:
```suggestion
    print(res.stderr, file=sys.stderr)
```